src/fireEvent.js

```diff
@@ -12,10 +34,14 @@ const findEventHandler = (
   const handler = getEventHandler(element, eventName);
   const hasHandler = handler != null || hasDescendandHandler;
 
-  const isHostComponent = typeof element.type === 'string';
+  const isHostComponent =
+    typeof element.type === 'string' || isTextInputComponent(element);
```

Comment:
please revert this check. host components are always of string type

Reply:
74b8c8a28e8957324404b77615288060fc34f9fc